numba/tests/test_builtins.py

```diff
@@ -982,6 +982,60 @@ def test_pow_usecase(self):
             r = cres.entry_point(x, y)
             self.assertPreciseEqual(r, pow_usecase(x, y))
 
+    def _check_min_max(self, pyfunc):
+        cfunc = njit()(pyfunc)
+        expected = pyfunc()
+        got = cfunc()
+        self.assertPreciseEqual(expected, got)
+
+    def test_min_max_iterable_input(self):
+
+        @njit
+        def frange(start, stop, step):
+            i = start
+            while i < stop:
+                yield i
+                i += step
+
+        def sample_functions(op):
+            yield lambda: op(range(10))
+            yield lambda: op(range(4, 12))
+            yield lambda: op(range(-4, -15, -1))
+            yield lambda: op([6.6, 5.5, 7.7])
+            yield lambda: op([(3, 4), (1, 2)])
+            yield lambda: op(frange(1.1, 3.3, 0.1))
+            yield lambda: op([np.nan, -np.inf, np.inf, np.nan])
+            yield lambda: op([(1,), (1,), (1,)])
+
+        for fn in sample_functions(op=min):
+            self._check_min_max(fn)
+
+        for fn in sample_functions(op=max):
+            self._check_min_max(fn)
+
+    def test_min_max_supplemental(self):
```

Comment:
Is this covered already by https://github.com/numba/numba/blob/1bfb050f4d69806e7c3b86881f51f21cfbf5ead3/numba/tests/test_parfors.py#L1032-L1084 ?

Reply:
It is - I put this test in whilst I was debugging and working through the code.  I intend to take it back out before un-WIP-ing...

Reply:
No worries, was just checking to save de-duplicating later :)